.github/workflows/check-pr.yml

```diff
@@ -26,3 +29,148 @@ jobs:
+        with:
+          token: ${{ secrets.WORKFLOW_TRIGGER_SECRET }}  # or use a custom token with proper permissions
+          repository: rdkcentral/firebolt-apis 
+          event-type: trigger-workflow
+          client-payload: '{"OPENRPC_PR_BRANCH": "${{ github.event.pull_request.head.ref }}"}'
+
+      - name: Store the workflow run ID
+        run: |
+          echo "Waiting for Repo2 to finish execution"
+  wait-for-sdk-generation:
+    needs: release
+    runs-on: ubuntu-latest
+    steps:
+      - name: Poll Firebolt-api for JavaScript SDK generation
+        id: poll-javascript-sdk
+        run: |
+          TOKEN="${{ secrets.WORKFLOW_TRIGGER_SECRET }}"
+          REPO_OWNER="rdkcentral"
+          REPO_NAME="firebolt-apis"
+          WORKFLOW_NAME="MFOS standalone sanity report - CORE,MANAGE,DISCOVERY"
+
+          while true; do
+            response=$(curl -s -H "Authorization: token $TOKEN" \
+              "https://api.github.com/repos/$REPO_OWNER/$REPO_NAME/actions/workflows" | jq \
+              --arg WORKFLOW_NAME "$WORKFLOW_NAME" \
+              '.workflows[] | select(.name == $WORKFLOW_NAME)')
+            
+            workflow_id=$(echo $response | jq -r '.id')
+            if [ "$workflow_id" == "null" ]; then
+              echo "Workflow not found, retrying in 10 seconds..."
+              sleep 10
+              continue
+            fi
+
+            # Get the latest workflow run
+            run_response=$(curl -s -H "Authorization: token $TOKEN" \
+              "https://api.github.com/repos/$REPO_OWNER/$REPO_NAME/actions/workflows/$workflow_id/runs?per_page=1")
+
+            status=$(echo $run_response | jq -r '.workflow_runs[0].status')
+            conclusion=$(echo $run_response | jq -r '.workflow_runs[0].conclusion')
+            
+            # Store conclusion if available
+            echo "JS_SDK_CONCLUSION=$conclusion" >> $GITHUB_ENV
+            
+            if [ "$status" == "completed" ] && [ "$conclusion" == "success" ]; then
+              echo "JavaScript SDK generated successfully for the OpenRPC changes."
+              break
+            elif [ "$status" == "completed" ] && [ "$conclusion" == "failure" ]; then
+              echo "Failed to generate JavaScript SDK for the OpenRPC changes."
+              exit 1
+            else
+              echo "Still in progress. Checking again in 120 seconds..."
+              sleep 120
+            fi
+          done
+      - name: Poll Firebolt-api for CPP SDK generation
+        id: poll-cpp-sdk
+        run: |
+          TOKEN="${{ secrets.WORKFLOW_TRIGGER_SECRET }}"
+          REPO_OWNER="rdkcentral"
+          REPO_NAME="firebolt-apis"
+          PR_NUMBER="${{ github.event.pull_request.number }}"
+          WORKFLOW_NAME="CXX build"
+
+          # Poll for the latest workflow run in Repo2
+          while true; do
+            # Fetch the list of workflows
+            response=$(curl -s -H "Authorization: token $TOKEN" \
```

Comment:
Same question as above regarding if we got the right workflow or not.

Reply:
It gets the workid of the workflow that is trigged by openRPC.Immediately after triggering the workflow we're getting the work flow id from the response.